src/test/java/io/vertx/mqtt/test/client/MqttClientOutOfOrderAcksTest.java

```diff
@@ -0,0 +1,213 @@
+      async.countDown();
+    });
+
+    client.connect(1883, "localhost", c -> {
+
+      // publish QoS = 1 message three times
+      for (int i = 0; i < 3; i++)
+        client.publish(MQTT_TOPIC,
+          Buffer.buffer(MQTT_MESSAGE.getBytes()),
+          MqttQoS.AT_LEAST_ONCE,
+          false,
+          false, h -> log.info("[CLIENT] publishing message id = " + h.result()));
+    });
+
+    async.await();
+    client.disconnect();
+  }
+
+  @Test
+  public void publishQoS2OutOfOrderAcks(TestContext context) throws InterruptedException {
+    Async serverStarted = context.async();
+
+    // set up server logic
+    server.endpointHandler(e -> {
+      log.info("[SERVER] Client connected");
+
+      e.publishHandler(p -> {
+        log.info("[SERVER] Received PUBLISH with message id = " + p.messageId());
+
+        // when received last PUBLISH message to acknowledge
+        if (p.messageId() == 3) {
+          // send PUBREC in order 3 --> 2 --> 1 -->
+          e.publishReceived(3);
+          e.publishReceived(2);
+          e.publishReceived(1);
+        }
+      });
+
+      e.publishReleaseHandler(pr -> {
+        log.info("[SERVER] Receive PUBREL with message id = " + pr);
+
+        // when received last PUBREL message to acknowledge
+        if (pr == 1) {
+          // send PUBREL in order 2 --> 1 --> 3 -->
+          e.publishComplete(2);
+          e.publishComplete(1);
+          e.publishComplete(3);
+        }
+      });
+
+      e.disconnectHandler(d -> log.info("[SERVER] Client disconnected"));
+      e.accept(false);
+    }).listen(ar -> {
+      if (ar.succeeded()) {
+        log.info("[SERVER] MQTT server listening on port " + ar.result().actualPort());
+        serverStarted.countDown();
+      } else {
+        log.error("[SERVER] Error starting MQTT server", ar.cause());
+        System.exit(1);
+      }
+    });
+    serverStarted.await();
+    
+    Async async = context.async(3);
+    MqttClient client = MqttClient.create(vertx);
+
+    Queue<Integer> expectOrder = new LinkedList<>();
+    // order we expect to receive acknowledgment for published message
+    expectOrder.add(2);
+    expectOrder.add(1);
+    expectOrder.add(3);
+
+
+    client.publishCompletionHandler(h -> {
+      context.assertTrue(h.intValue() == expectOrder.poll());
+      log.info("[CLIENT] Publish completed for message with id: " + h);
+      async.countDown();
+    });
+
+    client.connect(1883, "localhost", c -> {
+
+      // publish QoS = 2 message three times
+      for (int i = 0; i < 3; i++)
+        client.publish(MQTT_TOPIC,
+          Buffer.buffer(MQTT_MESSAGE.getBytes()),
+          MqttQoS.EXACTLY_ONCE,
+          false,
+          false, h -> log.info("[CLIENT] publishing message id = " + h.result()));
+    });
+
+    async.await();
+    client.disconnect();
+  }
+
+
+  @Before
+  public void before() {
+    MqttServerOptions serverOptions = new MqttServerOptions();
+    server = MqttServer.create(vertx, serverOptions);
```

Comment:
You can use the other `create` overload with no options as parameter (currently you are passing a default options that is what the other overload does).